File: PART1/07_01. memorydbToJpa/demo/src/main/java/com/example/demo/user/db/UserRepository.java
package com.example.demo.user.db;

import com.example.demo.user.model.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface UserRepository extends JpaRepository<UserEntity, Long> {

    // select * from user where score >= [??]

    List<UserEntity> findAllByScoreGreaterThanEqual(int sc);

    // select * from user where score >= ?? AND score <= ??
    List<UserEntity> findAllByScoreGreaterThanEqualAndScoreLessThanEqual(int min, int max);

    @Query(
        value = "select * from user as u where u.score >= :min AND u.score <= :max",
        nativeQuery = true
    )
    List<UserEntity> score(
        @Param(value = "min") int min,
        @Param(value = "max") int max);

}

File: PART3/ch03/ch03_springcloudapigateway/ch03_03/service/db/src/main/kotlin/org/delivery/db/user/UserRepository.kt
package org.delivery.db.user

import org.delivery.db.user.enums.UserStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository: JpaRepository<UserEntity, Long> {
    // select * from user where id = ? and status = ? order by id desc limit 1
    fun findFirstByIdAndStatusOrderByIdDesc(userId: Long?, status: UserStatus?): UserEntity?

    // select * from user where email = ? and password = ? and status = ? order by id desc limit 1
    fun findFirstByEmailAndPasswordAndStatusOrderByIdDesc(
        email: String?,
        password: String?,
        status: UserStatus?
    ): UserEntity?

}